internal/ls/lsutil/userpreferences_test.go

```diff
@@ -0,0 +1,87 @@
+package lsutil
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/go-json-experiment/json"
+	"github.com/microsoft/typescript-go/internal/modulespecifiers"
+	"gotest.tools/v3/assert"
+)
+
+func fillNonZeroValues(v reflect.Value, path string) {
+	t := v.Type()
+	for i := range t.NumField() {
+		field := v.Field(i)
+		fieldType := t.Field(i)
+		if !field.CanSet() {
+			continue
+		}
+		fieldPath := path + "." + fieldType.Name
+		switch field.Kind() {
+		case reflect.Bool:
+			field.SetBool(true)
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			field.SetInt(1)
+		case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+			field.SetUint(1)
+		case reflect.String:
+			val := getValidStringValue(field.Type(), fieldPath)
+			field.SetString(val)
+		case reflect.Slice:
+			if field.Type().Elem().Kind() == reflect.String {
+				field.Set(reflect.ValueOf([]string{"test"}))
+			}
+		case reflect.Struct:
+			fillNonZeroValues(field, fieldPath)
+		}
+	}
+}
+
+func getValidStringValue(t reflect.Type, path string) string {
```

Comment:
`path` doesn't seem to be used - maybe you needed it before? If you don't, you can get rid of `fieldPath` too above.